src/Swarm/Game/State.hs

```diff
@@ -269,11 +272,35 @@ makeLenses ''Notifications
 defaultRobotStepsPerTick :: Int
 defaultRobotStepsPerTick = 100
 
+-- | Type for remebering which robots were run in a robot step mode.
+--
+-- Note that it is only safe to transition to 'WorldTick' from 'SBefore'.
+--
+-- @
+--                     tick
+--     ┌────────────────────────────────────┐
+--     │                                    │
+--     │               step                 │
+--     │              ┌────┐                │
+--     │              ▼    │                │
+-- ┌───┴───┐ step  ┌───────┴───┐ step  ┌────┴─────┐
+-- │SBefore├──────►│SSingle RID├──────►│SAfter RID│
+-- └───────┘       └───────────┘       └────┬─────┘
+--                      ▲                   │
+--                      └───────────────────┘
+--                         view RID > oldRID
+-- @
+data SingleStep = SBefore | SSingle RID | SAfter RID
```

Comment:
This is better but I am still having trouble understanding exactly what this means, I think it still needs a lot more comments.
- Does the `SingleStep` value tell us which robot(s) should be stepped *next*, or which ones we just finished stepping?
- Why do `SSingle` and `SAfter` contain an `RID` but `SBefore` does not?  What do those `RID` values mean?
- Why is it only safe to transition to `WorldTick` from `SBefore`?